### PR TITLE
Integrate drop gem with backend

### PR DESCRIPTION
**Pull request by sebastiantoh**
- Currently supports only uploading 1 attachment. Attachments are stored as byte array. Likely will need to do some conversion if we were to display it.

TODO:
- Currently, the lat/lng of the dropped gems are some random hardcoded values. Will replace this with the actual current latlng
- Friend list is filled iwht moc kdata

### CODE REVIEW

File: backend/internal/adapters/secondary/repositories/gem.go [hunk 1]
```diff
@@ -52,10 +52,18 @@ func (r *gemRepository) GetPendingCollectionByUser(userId int64) ([]gems.Gem, er
...
+	// Necessary to use prepared statements here due to a bug in how goqu parses
+	// bytes (used for the gem attachments)
+	// More info here: https://github.com/doug-martin/goqu/issues/254
+	sql, args, _ := qb.Insert("gems").Prepared(true).Rows(gem).Returning("id").ToSQL()
 
 	var gemId int64
-	err := r.db.QueryRow(sql).Scan(&gemId)
+	stmt, err := r.db.Prepare(sql)
+	if err != nil {
+		return -1, fmt.Errorf("failed to prepare statement: %w", err)
```
Comment (ianyong) [inline backend/internal/adapters/secondary/repositories/gem.go]:
```suggestion
		return 0, fmt.Errorf("failed to prepare statement: %w", err)
```

Convention is to return the zero value which is `0` for `int64`.

**Comment by okg00dbye** [comment]
Can replace hard-coded latlng now that #44 is merged ~

**Review by ianyong — CHANGES_REQUESTED** [review]

File: backend/internal/adapters/secondary/repositories/gem.go (hunk 1 above)
Comment (sebastiantoh) [reply backend/internal/adapters/secondary/repositories/gem.go]: ok! does this apply to line 68 in the same file as well?

Comment (ianyong) [reply backend/internal/adapters/secondary/repositories/gem.go]: Ah yup! Thanks.

File: backend/internal/adapters/secondary/repositories/querybuilder.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package repositories
 
 import (
 	"github.com/doug-martin/goqu/v9"
+	_ "github.com/doug-martin/goqu/v9/dialect/postgres"
```
Comment (ianyong) [inline backend/internal/adapters/secondary/repositories/querybuilder.go:5]: Is this import needed?

Comment (sebastiantoh) [reply backend/internal/adapters/secondary/repositories/querybuilder.go:5]: Yep! 
image

**Review by ianyong — APPROVED** [review]
LGTM!

**Pull request merged**